### PR TITLE
Add gettext compatible paths to the list of translation folders

### PR DESCRIPTION
**Pull request by bancer**
This is probably the simplest solution to the problem described in #17887. 

If this is okay I could further optimize the code in translationsFolders().

### CODE REVIEW

**Comment by bancer** [comment]
@ADmad?

**Comment by ADmad** [comment]
The changes seem fine.

> If this is okay I could further optimize the code in translationsFolders().

Do you intended to add additional commits to this PR or make a separate PR?

**Review by LordSimal — APPROVED** [review]

**Comment by bancer** [comment]
> Do you intended to add additional commits to this PR or make a separate PR?

I will commit to the same branch. If this will be merged by that time I will create a new pull request.

File: src/I18n/MessagesFileLoader.php [hunk 1]
```diff
@@ -167,20 +171,12 @@ public function translationsFolders(): array
         if (!$localePaths && defined('APP')) {
```
Comment (ADmad) [inline src/I18n/MessagesFileLoader.php]: I wonder why the existence of `APP` is checked here when the constant used is `ROOT`. Seems like an oversight, would be a good idea to fix this too.

File: tests/TestCase/I18n/MessagesFileLoaderTest.php [hunk 2]
```diff
@@ -88,12 +88,12 @@ public function testTranslationFoldersSequence(): void
 
         $expected = [
             ROOT . DS . 'tests' . DS . 'test_app' . DS . 'resources' . DS . 'locales' . DS . 'en_' . DS,
```
Comment (ADmad) [inline tests/TestCase/I18n/MessagesFileLoaderTest.php]: Looking under `en_` doesn't make sense. We should skip using the value generated by `implode('_', [$locale['language'], $locale['region']]);` if the `region` is empty.

File: src/I18n/MessagesFileLoader.php [hunk 3]
```diff
@@ -157,29 +157,32 @@ public function translationsFolders(): array
         $locale = Locale::parseLocale($this->_locale) + ['region' => null];
 
         $folders = [
-            implode('_', [$locale['language'], $locale['region']]),
             $locale['language'],
+            // gettext compatible paths, see https://www.php.net/manual/en/function.gettext.php
+            $locale['language'] . DIRECTORY_SEPARATOR . 'LC_MESSAGES',
         ];
+        if (!empty($locale['region'])) {
```
Comment (ADmad) [inline src/I18n/MessagesFileLoader.php]:
```suggestion
        if ($locale['region']) {
```

File: src/I18n/MessagesFileLoader.php [hunk 4]
```diff
@@ -157,29 +157,32 @@ public function translationsFolders(): array
...
+            $languageRegion = implode('_', [$locale['language'], $locale['region']]);
+            $folders[] = $languageRegion;
+            // gettext compatible paths, see https://www.php.net/manual/en/function.gettext.php
+            $folders[] = $languageRegion . DIRECTORY_SEPARATOR . 'LC_MESSAGES';
```
Comment (markstory) [inline src/I18n/MessagesFileLoader.php:168]: Should we have tests for regional dialects?

Comment (bancer) [reply src/I18n/MessagesFileLoader.php:168]: testTranslationsFoldersGettextCompatible has been added.

**Review by ADmad — APPROVED** [review]

**Pull request merged**